internal/auth/oidc/repository_auth_method_update.go

```diff
@@ -17,32 +31,561 @@ import "context"
 // Options supported:
 //
 // * WithDryRun: when this option is provided, the auth method is retrieved from
-// the repo, updated based on the fieldMask, tested via Repository.TestAuthMethod
-// and any errors reported.  The updates are not peristed to the repository.
+// the repo, updated based on the fieldMask, tested via Repository.TestAuthMethod,
+// the results of the update are returned, and and any errors reported.  The
+// updates are not peristed to the repository.
 //
 // * WithForce: when this option is provided, the auth method is persistented in
 // the repository without testing it fo validity with Repository.TestAuthMethod.
 //
 // Successful updates must invalidate (delete) the Repository's cache of the
 // oidc.Provider for the AuthMethod.
-func (r *Repository) UpdateAuthMethod(ctx context.Context, m *AuthMethod, version uint32, fieldMaskPaths []string, _ ...Option) (*AuthMethod, error) {
+func (r *Repository) UpdateAuthMethod(ctx context.Context, am *AuthMethod, version uint32, fieldMaskPaths []string, opt ...Option) (*AuthMethod, int, error) {
 	const op = "oidc.(Repository).UpdateAuthMethod"
-	panic("to-do")
```

Comment:
Maybe leave this in?

Reply:
I panic about all the things these days.